### ISSUE TITLE
tab for indent

### ISSUE THREAD

**Issue by SamMaxwell** [fraywing/textAngular#939]
Is there a handy way to disable the new "tab key for indent" feature? It is driving my users nuts as they are trying to tab through the elements of the interface and one of the elements is wrapped with textAngular.

**Comment by JoelParke** [comment]
<p>See the https://github.com/fraywing/textAngular/wiki/Setting-Defaults where you can completely disable a particular key.<br/>This should solve your issue.</p>

**Comment by SamMaxwell** [comment]
I really did try to poke around to find this first. Sorry I asked a silly question.

**Comment by SamMaxwell** [comment]
So I got the indent and outdent tools to ignore the TabKey and the ShiftTabKey by commenting out the commandKeyCode in the texAngularSetup.js (I tried using testForKey: function(event) { return false; }, but that didn't work).

Now my question is ... how can I get the focus to move to the next/previous tab-able field in the form?

My users don't want to use mice.

**Comment by tholene** [comment]
I opened #970 with no luck, but just found that disabling the keys as you did works wonderfully with my directive. I'm still testing this, but so far there have been no errors! 

This is the directive I'm using:
````
link: function(scope, element, attributes) {
    // Timeout to prevent taTextElem from being null
    $timeout(function() {
        var taTextElem = element[0].querySelector('div[id^=taTextElement]');
        
        taTextElem.setAttribute('tabindex', attrs.myTabindex);

        attributes.$observe('myTabindex', function(value) {
            taTextElem.setAttribute('tabindex', value);        
        });
    });
}
````
This works fine with 
````
<text-angular ... myTabindex={{expression that evaluates into a number}}>
````

Note that in my case, I want to also update the tabindex whenever the value of myTabindex changes. That might not be your case, so you could just do:
````
link: function(scope, element, attributes) {
    // Timeout to prevent taTextElem from being null
    $timeout(function() {
       var taHTMLElem = element[0].querySelector('textArea[id^=taHtmlElement]');
                
       element[0].removeAttribute('tabindex');
       taHTMLElem.removeAttribute('tabindex');
    });
}
````

It seems that putting tabindex on ````<text-angular>```` also puts it on the 
````<textArea id=taHtmlElement...>````, along with the ````<div id=taTextElement...>````. 

I only want it to be on the ````taTextElement````, so I remove it from ````element[0]```` and the ````taHtmlElement````.

I hope this helps!

**Comment by atishjain29** [comment]
I want to move from one textangular element to another using tab key. Is there a way ?

**Comment by etholen** [comment]
Yes, look at my previous reply to see how this could be achieved. The standard tab index attribute works well but it gets applied twice, which is why I'm using a directive such as the one above. You'll also have to disable the tab key in textAngularSetup.js.

**Comment by YuantongL** [comment]
@SamMaxwell were there any luck you can tab to another textfield?

**Comment by SamMaxwell** [comment]
Not that I know of. We are switching over to using Kendo UI.

**Comment by YuantongL** [comment]
@tholene Where (in which file) actually you put the directive code?

**Comment by tholene** [comment]
Hi,
 
I only provided the link function in my reply. To connect back to the example, you could try to create a new directive either as a new file or in an existing file.

Something like this:

// myTabIndex.directive.js
var app = angular.module(‘myApp’,[]);

app.directive(‘myTabIndex’, myTabIndex);

function myTabIndex() {
	var directive = {
		restrict: ‘A’,
		link: linkFn
	};
	return directive;
}

function linkFn(scope, element, attrs) {…}

You don’t need all of this for the directive though, I have just defined the link function in my case.

Then you can just apply this directive to the <text-angular> element.

Hope it helps!
